### PR TITLE
Test Enabling Vuln Scan With BoringSSL

### PR DESCRIPTION
**Pull request by sealesj**
In an effort to enable vulnerability scanning on third_party dependencies, adding the boringssl directory with METADATA and OWNERS files should step towards all requirements on go/vuln-monitoring to get scanning enabled. I followed the instructions at go/thirdparty/metadata for creating this file, and I read http://go/chromium3p as an example for where to put these files and how to structure them.

This addresses b/230824334 but there is not an issue open currently.

Currently, only jseales and godofredoc listed on METDATA notify and OWNERS for testing 

Date used in METADATA field comes from 87f316d7748268eb56f2dc147bd593254ae93198 commit hash (current BoringSSL commit hash version in DEPS file) date

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by godofredoc — APPROVED** [review]
LGTM but let's get @chinmaygarde or @dnfield comments about this PR.

**Comment by godofredoc** [comment]
Seems like the licenses tools is expecting a copy of the license in the third party folder.

File: third_party/boringssl/OWNERS [hunk 1]
```diff
@@ -0,0 +1,2 @@
+jseales
+godofredoc
```
Comment (dnfield) [inline third_party/boringssl/OWNERS:2]: nit: newline at EOF

Are these really the right owners? I think @zanderso should be in here at least.

File: third_party/boringssl/METADATA [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  security {
+    # https://nvd.nist.gov/products/cpe/search/results?namingFormat=2.3&keyword=cpe%3A2.3%3Aa%3Agoogle%3Aboringssl
+    tag: "NVD-CPE2.3:cpe:/a:google:boringssl:2021-09-20"
+  }
+}
+# Notify reviewers and mailing lists
+presubmit: {
+  review_notify: "jseales"
+  review_notify: "godofredoc"
+}
```
Comment (dnfield) [inline third_party/boringssl/METADATA:26]: newline at eOF

**Comment by dnfield** [comment]
What will update this file?

We pull BSSL from Dart. Should this really be done at the Dart level instead?

File: third_party/boringssl/LICENSE [hunk 3]
```diff
@@ -0,0 +1,221 @@
+BoringSSL is a fork of OpenSSL. As such, large parts of it fall under OpenSSL
```
Comment (dnfield) [inline third_party/boringssl/LICENSE:1]: This is not the right license for this directory.... The files in this directory are just the METADATA and owners files.

We should instead update the license tool to ignore this directory.

**Comment by chinmaygarde** [comment]
I believe there was (or will be) a conversation about this. @dnfield Any updates?

**Pull request closed without merging**